test/functional/tag_controller_test.rb

```diff
@@ -143,41 +159,39 @@ def setup
   end
 
   test 'tag show range' do
-    get :show, id: tags(:spectrometer).name,
+    get :show, params: { id: tags(:spectrometer).name,
                start: (Time.now - 1.day).strftime('%d-%m-%Y'),
-               end: Time.now.strftime('%d-%m-%Y')
+               end: Time.now.strftime('%d-%m-%Y') }
 
     assert :success
     assert_not_nil :tags
   end
 
   test 'tag show JSON' do
-    get :show, id: tags(:spectrometer).name, format: 'json'
+    get :show, params: { id: tags(:spectrometer).name, format: 'json' }
 
     assert :success
     assert_not_nil :tags
-
     json = ActiveSupport::JSON.decode(@response.body)
-
     assert_not_nil json
     assert !assigns['notes'].empty?
     node = Node.find tags(:spectrometer).nodes.first.nid
-    assert_equal node.nid,                  json.first['nid']
+    assert_equal node.nid,                  json.first['node']['nid']
```

Comment:
I'm wondering if any JavaScript relied on `json.first['nid']` here -- did it not work like that? I had guessed this was a NodeTag record so maybe it ought to work as just `['nid']`?

Reply:
yeah it was not working with just ['nid'].

Reply:
I think we should look at the tagging systems to be sure they don't rely on this.

Reply:
@Souravirus please revert this back to how it was then we can talk on slack how to get it working, honestly i didn't take a deeper look at this because i know Rails's behaviour with hashes/arrays changes. Revert and let's see what problems it cause so we fix them instead.

Reply:
yeah ok reverting it back